### PR TITLE
Detect if the user is not logged in and abort the download.

### PR DESCRIPTION
**Pull request by bohning**

### CODE REVIEW

File: src/usdb_syncer/song_loader.py [hunk 1]
```diff
@@ -82,8 +82,10 @@ class Context:
     @classmethod
     def new(
         cls, details: SongDetails, options: Options, info: DownloadInfo, logger: Log
-    ) -> Context:
-        txt_str = usdb_scraper.get_notes(details.song_id, logger)
+    ) -> Union[Context, None]:
```
Comment (RumovZ) [inline src/usdb_syncer/song_loader.py:85]: In Python 3.10 you generally don't need `Union`. You can write `Context | None` instead.

File: src/usdb_syncer/usdb_scraper.py [hunk 2]
```diff
@@ -410,9 +410,11 @@ def get_notes(song_id: SongId, logger: Log) -> str:
...
+    text, success = _parse_song_txt_from_txt_page(soup)
+    return text, success
 
 
-def _parse_song_txt_from_txt_page(soup: BeautifulSoup) -> str:
-    return soup.find("textarea").string  # type: ignore
+def _parse_song_txt_from_txt_page(soup: BeautifulSoup) -> tuple[str, bool]:
+    if textarea := soup.find("textarea"):
+        return textarea.string, True  # type: ignore
+    return "", False
```
Comment (RumovZ) [inline src/usdb_syncer/usdb_scraper.py:420]: Can't we return `None` or an empty string in the error case here? I find that more readable and ergonomic than types you have to unpack first.

**Review by RumovZ — APPROVED** [review]
👍

File: src/usdb_syncer/song_loader.py (hunk 1 above)
Comment (bohning) [reply src/usdb_syncer/song_loader.py:85]: Thanks for the suggestion, I didn’t know that.

File: src/usdb_syncer/usdb_scraper.py (hunk 2 above)
Comment (bohning) [reply src/usdb_syncer/usdb_scraper.py:420]: Always appreciate your improvement suggestions!

**Comment by bohning** [comment]
I was wondering if there is a way to stop all queued downloads once the status of a missing login is determined.

**Comment by RumovZ** [comment]
It's possible, but I think it would be messy. Maybe instead you could check before any download is started at all, i.e. when the button is clicked?

**Comment by bohning** [comment]
Not sure if I did it the best way, but it seems to be working well.

**Comment by RumovZ** [comment]
Looks good. 🙂

**Pull request merged**